NewGui and Localization

PiezPiedPy opened the pull request:
So finally Trajectories gets to use the Dialogue system used by KSP itself, previously the Gui was using a system developed by Unity for debugging purposes.
* You can switch between the old and new Gui's if you wish with the toggle on the settings page.
* The Descent profile's default has been changed to Retrograde.
* Manual target entry text is now persistent.

Also Localization for the other languages supported by KSP has been included but with these early days of the new Gui there will be a few teething problems.
* Languages other than English may not display correctly as of yet, such things as text wrapping will probably occur with button text etc.
* Screenshots from users of languages other than English would be very welcome to help with sizing of things for the other languages.
* I have used Google Translate so expect text not to be translated properly, and I apologize if there are any naughtiness in the text.

I would appreciate it if people could download the test build and let me know how they get on, preferably before we merge. 😉
https://github.com/PiezPiedPy/KSPTrajectories/releases/tag/v1.7.1-pre.NewGUI-test

fat-lobyte commented:
Hi! I took a look at your GUI and it looks pretty amazing :)

I have a few comments though:

 * I think we should rename "Performance" to "Calculation Time"
 * Is the "Show Performance on Info Tab" check box really needed? I think we can just leave the performance counter in the Settings Tab.
 * Showing the Aerodynamic Model is not very important in the Info Tab. Maybe we should move that to the Settings Tab as well?

Generally, I think we should do something about the layout, as it is, it doesn't look very clean.
For example, most readouts in the Info and Target tab just look like a bunch of text. We should somehow separate out the different values into different "fields".

I think aligning the text and the numbers would help (kind of similar to how it's done in Kerbal Engineer).

The reason that "Target Distance" is a string and formatted the way that it is, is that I wanted to emulate having 3 fields in GUI. With the NewGUI, this emulation is not necessary anymore, we can just have 3 fields instead.

What do you think about that?

I will also take a look at the timer issue, because I think that thing would be very useful to have.

PiezPiedPy commented:
> * I think we should rename "Performance" to "Calculation Time"

I think so too, I'll change it.

> * Is the "Show Performance on Info Tab" check box really needed? I think we can just leave the performance counter in the Settings Tab.
> * Showing the Aerodynamic Model is not very important in the Info Tab. Maybe we should move that to the Settings Tab as well?

I put them on the info tab basically because the space was there, trivial matter to move them to settings.

> Generally, I think we should do something about the layout, as it is, it doesn't look very clean.

Would something more like the Profiler display be ok ?

> I will also take a look at the timer issue, because I think that thing would be very useful to have.

Thanks 😉

PiezPiedPy commented:
@fat-lobyte Integration Step slider adjusts from 0.5 to 5.0 in 0.5 increments, is that ok or would you prefer different settings ?

fat-lobyte commented on Localization/TrajectoriesLocalization.cfg:
```diff
@@ -0,0 +1,199 @@
+        #autoLOC_Trajectories_ImpactTime = Time to impact: <<1>>
+        #autoLOC_Trajectories_UseBlizzyToolbar = Use Blizzy's toolbar - Will take effect after restart
+        #autoLOC_Trajectories_UseCache = Use cache
+        #autoLOC_Trajectories_UseCacheNote = Trajectory will be more precise when the cache is disabled but computation time will be higher. It is not recommended to keep it disabled unless your CPU can handle the load.
```
Hopefully obsolete with the new integrator.

fat-lobyte commented:
I fixed the time to impact display, as you've probably seen.

> Would something more like the Profiler display be ok ?

Yes, but I've been tinkering a bit: https://github.com/fat-lobyte/KSPTrajectories/commits/profiler-fix
I wanted to get the alignment right, but now it doesn't match up with the header and the rightmost values are not inside the scroll box. But I'll figure it out :)

> I put them on the info tab basically because the space was there, trivial matter to move them to settings.

I think we shouldn't squander with space right now. I was hoping to add "Time to atmospheric exit" or "Apoapsis after aerobraking" or something like that, so the less clutter we have now, the better ;)

> Integration Step slider adjusts from 0.5 to 5.0 in 0.5 increments, is that ok or would you prefer different settings ?

Preferrably from 0.1 to 5.0 in logarithmic steps, but if that's too complicated, how about linear 0.3 steps from 0.3 to 4.5?

PiezPiedPy commented:
> I fixed the time to impact display, as you've probably seen.

Nice one 😃 

I see your having fun with the Popup Layout system 😉
I will have a play with the data readouts. First I will try right justifying the values and see how it looks.

So far I've
* Removed Performance from info page and associated toggle button from settings page.
* Renamed Performance on settings page to Calculation time.
* Moved Aerodynamic model from info page to settings page.

> Preferably from 0.1 to 5.0 in logarithmic steps, but if that's too complicated, how about linear 0.3 steps from 0.3 to 4.5?

Logarithmic steps it is then.

I'll get on it probably tonight if not tomorrow. Got to fit new injectors in my car today for it's MOT test, it's running like a bag of 💩 lol

PiezPiedPy commented:
![screenshot 9](https://user-images.githubusercontent.com/17555434/34889423-d4a08036-f7c5-11e7-80b1-3ba3a70222bc.png)
![screenshot 10](https://user-images.githubusercontent.com/17555434/34889429-de8c76d6-f7c5-11e7-8275-523d0cf855a6.png)
![screenshot 11](https://user-images.githubusercontent.com/17555434/34889438-e1b105d4-f7c5-11e7-9afe-643c6b6e83e0.png)

fat-lobyte commented:
Wow, I have to say that this looks amazing! Well done 👍 I'll give it a test soon, then I think we can merge.

Hey, do you think there's a way to remain (code) compatibility with KSP 1.2.2? I assume that would require all the localization stuff to be able to be turned off, right?
I have started playing Realism Overhaul and must say that's quite fun, but I also realized just how stuck you are with old KSP versions due the humongous amounts of mods that aren't always updated. I wonder if we could backport the new Trajectories version to the older version.

Btw, what about the merge conflict? Could you resolve that or should I? As far as I can see it's just a few variables that you renamed in code that I changed. Should be no biggie.

PiezPiedPy commented:
Thanks 😎 

If we are to do a separate version for 1.2.2 I recommend we start a new branch for it.

Removing the Localization shouldn't be too hard, all we have to do is hard code the en-us strings into the places that the Localization.Format methods are called.

I'll sort the conflict its only a couple of small changes as you seen.

PiezPiedPy commented:
Logarithmic slider for the RK4 Integrator is done, also fixed the conflicts.

Test build here -> https://github.com/PiezPiedPy/KSPTrajectories/releases/tag/v1.7.1-pre.NewGUI-test_build_2

fat-lobyte commented:
Hey there! I wanted to say thanks again, this is a truly amazing contribution  ❤️ 

The GUI looks much, much better now. I think this definitely warrants a 2.0 release, so whenever you say you are ready, I think we're gonna release the beast.

PiezPiedPy commented:
Thanks a lot, It was fun getting to grips with the new Dialogue system 😎 

Before the next release I've made some changes to the API to catch NRE's and made a pull request, I would like to get those changes in with the next release if possible.
